### PR TITLE
Some performance improvements in Azure.Core for .Net 6.0

### PR DESCRIPTION
**Pull request by AlexanderSher**
Performance gains in .Net 6.0 is about 10%

### CODE REVIEW

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs [hunk 1]
```diff
@@ -238,28 +274,43 @@ internal static IEnumerable<HttpHeader> GetHeaders(HttpHeaders headers, HttpCont
...
         internal static bool RemoveHeader(HttpHeaders headers, HttpContent? content, string name)
         {
             // .Remove throws on invalid header name so use TryGet here to check
+#if NET6_0_OR_GREATER
+            if (headers.NonValidated.Contains(name) && headers.Remove(name))
+            {
+                return true;
+            }
+
+            return content is not null && content.Headers.NonValidated.Contains(name) && content.Headers.Remove(name);
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:290]: Is the `Contains` call necessary here, or is it faster to check existence first in NonValidated?
```suggestion
            return headers.Remove(name) || content is not null && content.Headers.Remove(name);
```

File: sdk/storage/Azure.Storage.Common/src/Azure.Storage.Common.csproj [hunk 2]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <TargetFrameworks>$(RequiredTargetFrameworks);net6.0</TargetFrameworks>
```
Comment (christothes) [inline sdk/storage/Azure.Storage.Common/src/Azure.Storage.Common.csproj:3]: Do we need this in the specific client libraries also?
Can we add this to the storage test projects?

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs (hunk 1 above)
Comment (AlexanderSher) [reply sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:290]: It depends on what we do more often - remove existing or non-existing header. If header is added, then yes - simple call of `headers.Remove` is faster. And header is missing, then `headers.NonValidated.Contains` validates faster than `headers.Remove`. Probably we can use your approach.

File: sdk/storage/Azure.Storage.Common/src/Azure.Storage.Common.csproj (hunk 2 above)
Comment (AlexanderSher) [reply sdk/storage/Azure.Storage.Common/src/Azure.Storage.Common.csproj:3]: I'm going to revert changes in `Azure.Storage` to minimize the size of the PR and send those as a separate one.

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs (hunk 1 above)
Comment (AlexanderSher) [reply sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:290]: It looks like `headers.Remove` still thrown on invalid header, so we must call  `headers.NonValidated.Contains` first

File: global.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "Microsoft.Build.Traversal": "3.0.23"
   },
   "sdk": {
-    "version": "6.0.100",
+    "version": "6.0.403",
```
Comment (christothes) [inline global.json:6]: It doesn't look like we've added net6.0 to the TFM for `Azure.Core.Tests.csproj` yet. I think we should add it

Comment (AlexanderSher) [reply global.json:6]: .net6.0 is added to all test projects: https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Directory.Build.Common.props#L123-L125

Comment (christothes) [reply global.json:6]: Oh, right 🤦 😄

**Review by christothes — APPROVED** [review]

**Pull request merged**